test/services/application/ApplicationServiceSpec.scala

```diff
@@ -76,6 +80,82 @@ class ApplicationServiceSpec extends UnitSpec with ExtendedTimeout {
+
+      when(appRepositoryMock.findByUserId(eqTo(userId), eqTo(frameworkId))).thenReturn(Future.successful(faststreamApplication))
+      when(evalPhase3ResultMock.getPassmarkEvaluation(eqTo(applicationId))).thenReturn(Future.successful(passmarkEvaluation))
+
+      val passedSchemes = underTest.getPassedSchemes(userId, frameworkId).futureValue
+
+      passedSchemes mustBe List(SchemeType.Commercial)
+
+    }
+
+    "retrieve passed schemes for Faststream application with fast pass approved" in new ApplicationServiceTest {
+
+      val userId = "userId"
+      val applicationId = "appId"
+      val frameworkId = ""
+
+      val faststreamApplication = ApplicationResponse(applicationId, "", ApplicationRoute.Faststream,
+        userId,ProgressResponse(applicationId, fastPassAccepted = true), None)
+
+      when(appRepositoryMock.findByUserId(eqTo(userId), eqTo(frameworkId))).thenReturn(Future.successful(faststreamApplication))
+      when(schemeRepositoryMock.find(eqTo(applicationId))).thenReturn(Future.successful(SelectedSchemes(List(SchemeType.Commercial),
+        orderAgreed = true, eligible = true)))
+
+      val passedSchemes = underTest.getPassedSchemes(userId, frameworkId).futureValue
+
+      passedSchemes mustBe List(SchemeType.Commercial)
+
+    }
+
+    "retrieve passed schemes for Edip application" in new ApplicationServiceTest {
+
+      val userId = "userId"
+      val applicationId = "appId"
+      val frameworkId = ""
+
+      val edipApplication = ApplicationResponse(applicationId, "", ApplicationRoute.Edip, userId, ProgressResponse(applicationId), None)
+      val passmarkEvaluation = PassmarkEvaluation("", None, List(SchemeEvaluationResult(SchemeType.Edip, "Green")))
+
+      when(appRepositoryMock.findByUserId(eqTo(userId), eqTo(frameworkId))).thenReturn(Future.successful(edipApplication))
+      when(evalPhase1ResultMock.getPassmarkEvaluation(eqTo(applicationId))).thenReturn(Future.successful(passmarkEvaluation))
+
+      val passedSchemes = underTest.getPassedSchemes(userId, frameworkId).futureValue
+
+      passedSchemes mustBe List(SchemeType.Edip)
+
+    }
+
+    "retrieve passed schemes for Sdip application" in new ApplicationServiceTest {
+
+      val userId = "userId"
+      val applicationId = "appId"
+      val frameworkId = ""
+
+      val sdipApplication = ApplicationResponse(applicationId, "", ApplicationRoute.Sdip, userId, ProgressResponse(applicationId), None)
+      val passmarkEvaluation = PassmarkEvaluation("", None, List(SchemeEvaluationResult(SchemeType.Sdip, "Green")))
+
+      when(appRepositoryMock.findByUserId(eqTo(userId), eqTo(frameworkId))).thenReturn(Future.successful(sdipApplication))
+      when(evalPhase1ResultMock.getPassmarkEvaluation(eqTo(applicationId))).thenReturn(Future.successful(passmarkEvaluation))
+
+      val passedSchemes = underTest.getPassedSchemes(userId, frameworkId).futureValue
+
+      passedSchemes mustBe List(SchemeType.Sdip)
+
+    }
   }
 
   trait ApplicationServiceTest {
```

Comment:
A bit pedantic, but there are no negative tests here to check that the failed schemes are filtered out.